packages/amazonq/src/lsp/chat/webviewProvider.ts

```diff
@@ -51,25 +40,60 @@ export class AmazonQChatViewProvider implements WebviewViewProvider {
         this.webview = webviewView.webview
 
         const lspDir = Uri.parse(LanguageServerResolver.defaultDir)
+        const dist = Uri.joinPath(globals.context.extensionUri, 'dist')
         webviewView.webview.options = {
             enableScripts: true,
             enableCommandUris: true,
-            localResourceRoots: [lspDir, Uri.parse(path.dirname(this.mynahUIPath))],
+            localResourceRoots: [lspDir, dist],
         }
 
+        const source = 'vue/src/amazonq/webview/ui/amazonq-ui-connector-adapter.js' // Sent to dist/vue folder in webpack.
+        const serverHostname = process.env.WEBPACK_DEVELOPER_SERVER
+        const connectorAdapterPath =
+            serverHostname !== undefined
+                ? Uri.parse(serverHostname)
+                      .with({ path: `/${source}` })
+                      .toString()
+                : webviewView.webview.asWebviewUri(Uri.parse(path.join(dist.fsPath, source))).toString()
         const uiPath = webviewView.webview.asWebviewUri(Uri.parse(this.mynahUIPath)).toString()
-        webviewView.webview.html = await this.getWebviewContent(uiPath)
+        webviewView.webview.html = await this.getWebviewContent(uiPath, connectorAdapterPath)
 
         this.onDidResolveWebviewEmitter.fire()
+        performance.mark(amazonqMark.open)
     }
 
-    private async getWebviewContent(mynahUIPath: string) {
+    private async getWebviewContent(mynahUIPath: string, hybridChatConnector: string) {
+        const featureConfigData = await featureConfig.getFeatureConfigs()
+
+        const isSM = isSageMaker('SMAI')
+        const isSMUS = isSageMaker('SMUS')
+        const disabledCommands = isSM ? `['/dev', '/transform', '/test', '/review', '/doc']` : '[]'
```

Comment:
Oh interesting, the legacy agents aren't available in SM? Is this related to SM having its own Auth logic (uses IAM for chat right?)?

Reply:
I think so, it was added when I was on vacation I think. This is the same string but from the current webview content: https://github.com/aws/aws-toolkit-vscode/blob/1c9103719f43f4adf1915873886f9f8ed51eed5d/packages/core/src/amazonq/webview/generators/webViewContent.ts#L86